Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,261 @@
+                bool success = float.TryParse(balances[i], out float number);
+                if (success) floatArr[i - failLength] = number;
+                else failLength++;
+            }
+            if(failLength == 0) return floatArr;
+            float[] failFloatArr = new float[floatArr.Length - failLength];
+            for (int i = 0; i < failFloatArr.Length; i++)
+            {
+                failFloatArr[i] = floatArr[i];
+            }
+            return failFloatArr;
+        }
+
+        /// <summary>
+        /// Takes an array of floats and returns a single float of the highest value.
+        /// </summary>
+        public static float HighestBalance(float[] balances)
+        {
+            float highestBalance = balances[0];
+            if (balances.Length == 0) return highestBalance;
+            for (int i = 1; i < balances.Length; i++)
+            {
+                if (balances[i] > highestBalance) highestBalance = balances[i];
+            }
+            return highestBalance;
+        }
+
+        /// <summary>
+        /// Takes an array of names and balances, then returns the highest value of all balances.
+        /// </summary>
+        /// <param name="peopleAndBalances"></param>
+        /// <returns></returns>
+        public static float HighestBalanceEver(string[] peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float highestBalanceEver = HighestBalance(StringArrToFloatArr(workingPeopleAndBalances[0][1..]));
+            if (workingPeopleAndBalances.Length == 1) return highestBalanceEver;
+            for (int i = 1 ; i < workingPeopleAndBalances.Length; i++)
+            {
+                float balanceToTest = HighestBalance(StringArrToFloatArr(workingPeopleAndBalances[i][1..]));
+                if (balanceToTest > highestBalanceEver)
+                {
+                    highestBalanceEver = balanceToTest;
+                }
+            }
+            return highestBalanceEver;
+        }
+
+        /// <summary>
+        /// Takes a balance then finds and returns all the names that share that balance
+        /// </summary>
+        /// <param name="balance"></param>
+        /// <param name="peopleAndBalances"></param>
+        /// <returns></returns>
+        public static string BalanceNames(float balance, string[]peopleAndBalances)
+        {
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            string totalNames = "";
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float[] balances = StringArrToFloatArr(workingPeopleAndBalances[i][1..]);
+                bool isBalance = false;
+                foreach (var item in balances)
+                {
+                    if (item == balance) isBalance = true;
+                }
+                if (isBalance) totalNames += $"{workingPeopleAndBalances[i][0]}, ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)